remoteappmanager/handlers/base_handler.py

```diff
@@ -40,3 +41,20 @@ def render(self, template_name, **kwargs):
 
         args.update(kwargs)
         super(BaseHandler, self).render(template_name, **args)
+
+    def write_error(self, status_code, **kwargs):
+        """Render error page for uncaught errors"""
+        status_message = responses.get(status_code, 'Unknown HTTP Error')
+        message = ""
+
+        # If this error was caused by an uncaught exception
+        # log exception message and reference number as well
+        exc_info = kwargs.get('exc_info')
+        if exc_info:
+            exception = exc_info[1]
+            ref = self.log.issue(status_message, exception)
+            reason = getattr(exception, 'reason', '')
+            message = '{} Ref.: {}'.format(reason, ref)
```

Comment:
You still open up to the chance that this code is actually presenting the user with internal information. It depends on the exception reason text, and if the exception text contains sensitive information for whatever reason, it will be shown to the user.